server.rb

```diff
@@ -26,19 +27,21 @@ def self.run
 
 while true
   proc {
-    Dir.chdir("/tmp") do
-      Dir.pwd
-      Process.pid
-      'hello'.multiply_vowels(3){ :ohai }
-      sleep rand*0.5
+    Dir.mktmpdir do |tmp|
```

Comment:
is this temp dir cleaned up ?

Reply:
Yep! I made sure of that. From the [docs](https://ruby-doc.org/stdlib-2.5.1/libdoc/tmpdir/rdoc/Dir.html#method-c-mktmpdir):

> If a block is given, it is yielded with the path of the directory. The directory and its contents are removed using FileUtils.remove_entry before ::mktmpdir returns.